спринт 10

disochkin opened the pull request.

disochkin committed:
спринт 10

disochkin committed:
some fixes

disochkin committed:
добавил логирование

disochkin committed:
code style fix

disochkin committed:
code style fix

disochkin committed:
code style fix

disochkin committed:
code style fix

disochkin committed:
fix pom.xml
remove redundant junit-jupiter-api

disochkin committed:
fix controllers

disochkin committed:
fix codestyle

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,92 @@
+            return new ResponseEntity<>(film, HttpStatus.OK);
+        } catch (Exception e) {
+            log.debug("Ошибка валидации запроса на создание нового фильма: {}", e);
+            return new ResponseEntity<>(film, HttpStatus.INTERNAL_SERVER_ERROR);
```
А зачем возвращать пользователю INTERNAL_SERVER_ERROR? 
Может быть лучше не перехватывать, а вернуть ошибку валидации?

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,110 @@
+            return new ResponseEntity<>(user, HttpStatus.OK);
+        } catch (Exception e) {
+            log.debug("Ошибка создания пользователя", e.toString());
+            return new ResponseEntity<>(user, HttpStatus.INTERNAL_SERVER_ERROR);
```
Комментарий такой же, что и для контроллера фильмов

disochkin replied:
Убрал ResponseEntity
но непонятно как вернуть текст ошибки, если потребуется

avfyodorov commented on src/test/java/ru/yandex/practicum/filmorate/UserModelTest.java:
```diff
@@ -0,0 +1,23 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class UserModelTest {
```
Хотелось бы, конечно, немного больше тестов. 🙂

disochkin replied:
Добавил тесты для моделей

avfyodorov reviewed (commented):
Добрый вечер! 

Хорошая работа, от меня всего пара уточнений.

disochkin committed:
исправил замечания
добавил тесты для моделей
логику валидации перенес в аннотации дял полей моделей

avfyodorov reviewed (commented):
Добрый вечер !

Замечаний нет.
Работа принята.
